src/ClpSolverInterface.jl

```diff
@@ -643,23 +757,45 @@ function MOI.get(m::ClpMOIModel,
     return m.nrows_interval
 end
 
+function MOI.get(m::ClpMOIModel,
+        ::MOI.NumberOfConstraints{MOI.ScalarAffineFunction{Float64},MOI.EqualTo{Float64}})
+    return m.nrows_equalto
+end
+
 # Optimization and additional attributes
 function MOI.optimize!(m::ClpMOIModel)
     initial_solve_with_options(m.inner,m.solveroptions)
-    m.solution = primal_column_solution(m.inner)
-    m.rowsolution = primal_row_solution(m.inner)
-    m.dualcolsolution = dual_column_solution(m.inner) .* get_obj_sense(m.inner)
-    m.dualrowsolution = dual_row_solution(m.inner) .* get_obj_sense(m.inner)
+    if is_proven_dual_infeasible(m.inner)
+        # m.solution = unbounded_ray(m.inner) # .* get_obj_sense(m.inner)
+    else
+        m.solution = primal_column_solution(m.inner)
+        m.rowsolution = primal_row_solution(m.inner)    
+    end
+    
+    if is_proven_primal_infeasible(m.inner)
+        # m.dualrowsolution = infeasibility_ray(m.inner) # .* get_obj_sense(m.inner)
+    else
+        m.dualrowsolution = dual_row_solution(m.inner) .* get_obj_sense(m.inner)
+        m.dualcolsolution = dual_column_solution(m.inner) .* get_obj_sense(m.inner)
+    end
+    
+    # @show MOI.get(m, MOI.ResultCount())
+    # @show primal_feasible(m.inner)
+    # @show dual_feasible(m.inner)
+    # @show is_proven_primal_infeasible(m.inner)
+    # @show is_proven_dual_infeasible(m.inner)
 end
 
 function MOI.get(m::ClpMOIModel, ::MOI.TerminationStatus)
     s = ClpCInterface.status(m.inner)
-    if s == 0
+    if s == 0 
         return MOI.Success
     elseif s == 1
```

Comment:
`InfeasibleNoResult` and `UnboundedNoResult` do not apply if a certificate is available. `s == 1` and `s == 2` should be `Success`.

Reply:
I agree with that. I actually have this commented in the code

```
# if s in [0, 1, 2] 
#     return MOI.Success
```

But I changed it because I can't find a method that returns the infeasibitliy certificate.

Reply:
https://github.com/JuliaOpt/Clp.jl/blob/33cb6c31e72bd958e409d3dcf868f1a8a16cec58/src/ClpCInterface.jl#L828

Reply:
Thanks